cmd/container-suseconnect/main.go

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2015 SUSE LLC. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Gives access to repositories during docker build and run using the host
+// machine credentials.
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	cs "github.com/SUSE/container-suseconnect/pkg/container-suseconnect"
+	"github.com/urfave/cli"
+)
+
+func main() {
+	// Set the basic CLI metadata
+	app := cli.NewApp()
+	app.Copyright = fmt.Sprintf("© %d SUSE LCC", time.Now().Year())
+	app.Name = "container-suseconnect"
+	app.Version = "2.0.0"
+	app.Usage = ""
+	app.UsageText =
+		`This application can be used to retrieve basic metadata about SLES
+   related products and module extensions.
+
+   Please use the 'list-products' subcommand for listing all currently
+   available products including their repositories and a short description.
+
+   Use the 'list-modules' subcommand for listing available modules, where
+   their 'Identifier' can be used to enable them via the ADDITIONAL_MODULES
+   environment variable during container creation/run.
+
+   The 'zypper' subcommand runs the application as zypper plugin and is only
+   intended to use for debugging purposes.`
+
+	// Switch the application behavior regarding the basename
+	switch filepath.Base(os.Args[0]) {
+	case "container-suseconnect-zypp":
+		app.Action = runZypperPlugin
```

Comment:
As far as I understand it, this is the only action that doesn't require parameters, right? We could probably show a warning if users pass an extra parameter or something like that. Just wondering because of the possible combinations between the action and the available commands.

Reply:
Yep, this is the default action which is executed if no arguments are provided to the application. It would be possible to hide the default action from the commands list, but my intention was to make the default action not hidden that users can see and execute all available commands.

I added some hint about the `default` action when running something like `container-suseconnect -h`. What do you think?

Reply:
Fair enough :+1: